src/ln/msgs.rs

```diff
@@ -518,6 +605,19 @@ pub enum HTLCFailChannelUpdate {
 	}
 }
 
+/// Meessages could have optional fields to use with extended features
+/// Due to serialization issues, with encapsutale in enum
+#[derive(Clone)]
+pub enum OptionalField<T> {
+	/// Optional field is included in message
+	Present {
```

Comment:
Why not just Present(T)?

Reply:
Thanks, still learning to be a good rustacean